gravitee-node-cache/gravitee-node-cache-plugin-redis/src/main/java/io/gravitee/node/plugin/cache/redis/RedisCache.java

```diff
@@ -0,0 +1,275 @@
+
+    private String getRedisEntryKey(String key) {
+        return this.name + SEPARATOR + key;
+    }
+
+    private Response throwExceptionOnError(Response response) {
+        if (ResponseType.ERROR == response.type()) {
+            throw new CacheException("Error during cache operation: " + response.format());
+        }
+        return response;
+    }
+
+    @Override
+    public Collection<V> values() {
+        throw new UnsupportedOperationException("Redis cache does not support values operation");
+    }
+
+    @Override
+    public Flowable<K> rxKeys() {
+        return Single
+            .fromCompletionStage(this.redisAPI.keys(getRedisEntryKey("*")).toCompletionStage())
+            .map(this::throwExceptionOnError)
+            .flattenStreamAsFlowable(Response::stream)
+            .map(response -> (K) response.toString(StandardCharsets.UTF_8).replace(getRedisEntryKey(""), ""));
+    }
+
+    @Override
+    public Set<K> keys() {
+        return this.rxKeys().collectInto(new HashSet(), (set, key) -> set.add(key)).blockingGet();
+    }
+
+    @Override
+    public Set<Map.Entry<K, V>> entrySet() {
+        throw new UnsupportedOperationException("Redis cache does not support entrySet operation");
+    }
+
+    @Override
+    public int size() {
+        throw new UnsupportedOperationException("Redis cache does not support size operation");
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return this.rxIsEmpty().blockingGet();
+    }
+
+    @Override
+    public Single<Boolean> rxIsEmpty() {
+        return this.rxKeys().isEmpty();
+    }
+
+    @Override
+    public boolean containsKey(K key) {
+        return this.rxContainsKey(key).blockingGet();
+    }
+
+    @Override
+    public Single<Boolean> rxContainsKey(K key) {
+        return Single
+            .fromCompletionStage(this.redisAPI.exists(List.of(key + SEPARATOR)).toCompletionStage())
+            .map(this::throwExceptionOnError)
+            .map(resp -> resp.toInteger() > 0);
+    }
+
+    @Override
+    public V get(K key) {
+        return this.rxGet(key).blockingGet();
+    }
+
+    @Override
+    public Maybe<V> rxGet(K key) {
+        return Maybe
+            .fromCompletionStage(this.redisAPI.get(getRedisEntryKey(key)).toCompletionStage())
+            .map(this::throwExceptionOnError)
+            .map(response -> (V) cacheConfiguration.getMapper().toValue(asString(response)));
+    }
+
+    @Override
+    public V put(K key, V value) {
+        return this.rxPut(key, value).blockingGet();
+    }
+
+    @Override
+    public Maybe<V> rxPut(K key, V value) {
+        return Maybe
+            .fromCompletionStage(
+                this.redisAPI.setnx(getRedisEntryKey(key), (String) cacheConfiguration.getMapper().toCachedValue(value)).toCompletionStage()
+            )
+            .map(this::throwExceptionOnError)
+            .map(r -> {
+                this.cacheListeners.values().forEach(listener -> listener.onEntryAdded(key, value));
```

Comment:
if you can / not too resource consuming, maybe try distinguish add from update, as the listener have two different methods.